Library/Homebrew/command.rb

```diff
@@ -72,13 +72,51 @@ def self.desc(desc)
       end
     end
 
+    def self.get_invalid_options(argv_options_only)
+      argv_options_only = argv_options_only.uniq
+      valid_options_with_values =
+        @valid_options
+        .select { |option_hash| !option_hash[:value].nil? }
```

Comment:
Instead of `!option_hash[:value].nil?`, you can just check `option_hash[:value]` and ruby will do the necessary truthy evaluation.